CommunicationLayer/MessagingFramework/MotorDetailsMessage.cpp

```diff
@@ -0,0 +1,153 @@
+    const int VOLTAGE_RAIL_SUPPLY_19_V = 41;
+    const int HEAT_SINK_TEMPERATURE = 45;
+    const int MOTOR_TEMPTERATURE = 49;
+    const int DSP_BOARD_TEMPERATURE = 53;
+    const int DC_BUS_AMP_HOURS = 57;
+    const int ODOMETER = 61;
+    const int SLIP_SPEED = 65;
+}
+
+MotorDetailsMessage::MotorDetailsMessage(const QByteArray& messageData, unsigned char motorNumber)
+: messageData_(messageData)
+, motorNumber_(motorNumber)
+{
+}
+
+float MotorDetailsMessage::phaseCCurrent() const
+{
+    return getFloat(messageData_, PHASE_CCURRENT);
+}
+
+float MotorDetailsMessage::phaseBCurrent() const
+{
+    return getFloat(messageData_, PHASE_BCURRENT);
+}
+
+float MotorDetailsMessage::motorVoltageReal() const
+{
+    return getFloat(messageData_, MOTOR_VOLTAGE_REAL);
+}
+
+float MotorDetailsMessage::motorVoltageImaginary() const
+{
+    return getFloat(messageData_, MOTOR_VOLTAGE_IMAGINARY);
+}
+
+float MotorDetailsMessage::motorCurrentReal() const
+{
+    return getFloat(messageData_, MOTOR_CURRENT_REAL);
+}
+
+float MotorDetailsMessage::motorCurrentImaginary() const
+{
+    return getFloat(messageData_, MOTOR_CURRENT_IMAGINARY);
+}
+
+float MotorDetailsMessage::backEmfReal() const
+{
+    return getFloat(messageData_, BACK_EMF_REAL);
+}
+
+float MotorDetailsMessage::backEmfImaginary() const
+{
+    return getFloat(messageData_, BACK_EMF_IMAGINARY);
+}
+
+float MotorDetailsMessage::voltageRailSuppply15V() const
+{
+    return getFloat(messageData_, VOLTAGE_RAIL_SUPPPLY_15_V);
+}
+
+float MotorDetailsMessage::voltageRailSupply33V() const
+{
+    return getFloat(messageData_, VOLTAGE_RAIL_SUPPLY_33_V);
+}
+
+float MotorDetailsMessage::voltageRailSupply19V() const
+{
+    return getFloat(messageData_, VOLTAGE_RAIL_SUPPLY_19_V);
+}
+
+float MotorDetailsMessage::heatSinkTemperature() const
+{
+    return getFloat(messageData_, HEAT_SINK_TEMPERATURE);
+}
+
+float MotorDetailsMessage::motorTempterature() const
+{
+    return getFloat(messageData_, MOTOR_TEMPTERATURE);
+}
+
+float MotorDetailsMessage::dspBoardTemperature() const
+{
+    return getFloat(messageData_, DSP_BOARD_TEMPERATURE);
+}
+
+float MotorDetailsMessage::dcBusAmpHours() const
+{
+    return getFloat(messageData_, DC_BUS_AMP_HOURS);
+}
+
+float MotorDetailsMessage::odometer() const
+{
+    return getFloat(messageData_, ODOMETER);
+}
+
+float MotorDetailsMessage::slipSpeed() const
+{
+    return getFloat(messageData_, SLIP_SPEED);
+}
+
+QString MotorDetailsMessage::toString() const
+{
+    QString messageString;
+    if (motorNumber_ == 0x0) {
+        messageString += QString::number(MessageDefines::Motor0Details) + ", ";
+    }
+    else {
+        // For the sake of not halting programs execution, treat all other cases as motor one
+        if (motorNumber != 0x1) {
+            // Log that this was used incorrectly
+            qDebug() << "Invalid motor number, defaulting to motor 1";
+        }
+        messageString += QString::number(MessageDefines::Motor1Details) + ", ";
+    }
```

Comment:
If the method of distinguishing the two motors is appropriate, is this how you would like to deal with an incorrect motor ID being input? If this is I will of course change it to compare against an enum instead, as this was just a mock up to see what you thought. 
Personally I would prefer to throw an exception in the constructor, as I suspect the only time an incorrect id should be passed into the constructor is when the logic to create the messages is faulty in the first place, and doing something like this might allow that faulty code to make it onto the car if it wasn't caught by my qDebug statement during testing. Let me know what you think is appropriate